components/DataLiberation/Tests/CSSTokenizerTest.php

```diff
@@ -0,0 +1,918 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+use WordPress\DataLiberation\URL\CSSTokenizer;
+
+/**
+ * Comprehensive CSS tokenizer tests based on the CSS Syntax Level 3 specification.
+ * Test corpus from @rmenke/css-tokenizer-tests
+ */
+class CSSTokenizerTest extends TestCase {
+
+	/**
+	 * Tests that the tokenizer produces the expected tokens for all test cases.
+	 *
+	 * @dataProvider corpus_provider
+	 */
+	public function test_tokenizer_matches_spec( string $css, array $expected_tokens ): void {
+		$processor = CSSTokenizer::create( $css );
+		$actual_tokens = $this->collect_tokens( $processor, $css );
+
+		// Compare token count first
+		$this->assertCount(
+			count( $expected_tokens ),
+			$actual_tokens,
+			'Token count mismatch for CSS: ' . var_export( $css, true )
+		);
+
+		// Compare each token
+		foreach ( $expected_tokens as $index => $expected_token ) {
+			if ( ! isset( $actual_tokens[ $index ] ) ) {
+				$this->fail( "Missing token at index $index for CSS: " . var_export( $css, true ) );
+			}
+			$actual_token = $actual_tokens[ $index ];
+			$this->assert_token_matches( $expected_token, $actual_token, $index, $css );
+		}
+	}
+
+	/**
+	 * Provides all test cases from the test corpus.
+	 *
+	 * @return array
+	 */
+	static public function corpus_provider(): array {
+		static $test_cases = null;
+
+		if ( null === $test_cases ) {
+			$test_cases = require __DIR__ . '/css-test-cases.php';
+		}
+
+		$data = array();
+		foreach ( $test_cases as $test_name => $test_case ) {
+			$data[ $test_name ] = array(
+				$test_case['css'],
+				$test_case['tokens'],
+			);
+		}
```

Comment:
not that it matters, but we can also use `yield $test_name => array( $test_case['css'], $test_case['tokens'] )`

though this makes me wonder more, why are we doing this at all? why are we unpacking the generated test cases to reassemble them in the exact same shape as we chose to represent them in the first place?

did you make these choices?